# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_quad.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/ps_quad.glsl
--- a/gfx/wr/webrender/res/ps_quad.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/ps_quad.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -6,11 +6,11 @@
 
 #include shared,rect,transform,render_task,gpu_buffer
 
-flat varying vec4 v_color;
-flat varying ivec2 v_flags;
+flat varying mediump vec4 v_color;
+flat varying mediump ivec2 v_flags;
 
 #ifndef SWGL_ANTIALIAS
-varying vec2 vLocalPos;
+varying highp vec2 vLocalPos;
 #endif
 
 #ifdef WR_VERTEX_SHADER